File: p4.py
# smallest 10000 and largest 999*999=998001

def check_palindrome(number):
    return str(number) == ''.join(reversed(str(number)))


largest = 0
for first in range(100,999):
    for second in range(100,999):
        product = first* second
        if check_palindrome(product) and product > largest:
            largest = product

print(largest)

File: Solutions/1-100/p4.rs
// Solution to problem 4 in request

fn is_palindrome(number: u64) -> bool {
    let mut reversed = 0;
    let mut copy = number;
    while copy > 0 {
        reversed *= 10;
        // add last digit
        reversed += copy % 10;
        // since copy is an integer, throws away anything after the ,
        copy /= 10;
    }
    number == reversed
}

fn main() {
    let mut largest: u64 = 0;
    //initialize temp variable without initial value
    let mut temp_product: u64;
    for first in 100..1000 {
        for second in 100..1000 {
            temp_product = first * second;
            if is_palindrome(temp_product) & (temp_product > largest) {
                largest = temp_product
            }
        }
    }
    println!("The answer is: {}", largest)
}
